zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1968,7 +1968,7 @@ public <T> T retryUntilConnected(final Callable<T> callable)
     if (_monitor != null) {
       _monitor.increaseOutstandingRequestGauge();
     }
-    int retryCount = 1;
+    int retryCount = 0;
```

Comment:
Thanks for working on the fix! 
Changing retryCount does it mean we start retry immediately after we get error?

Reply:
Earlier and now too, we retry. The only difference is after failure, how long do we wait. Prior to exponential back-off, we didn't wait at all. Now we wait for whatever is the time ExponentialBackOff strategy calculates.